### ISSUE TITLE
[Feature]: Shapes with blur background.

### ISSUE THREAD

**Issue by abhaybisht314ecorp** [fabricjs/fabric.js#10340]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

1. I want to make shapes with blurry background ( glassmorphic in nature ).

### Current State

1. There is now way of adding CSS styles to fabric shapes, that can help me to achieve this.
eg: filter: blur(10px); 
      backdrop-filter: blur(10px);

**Comment by asturur** [comment]
What is a shape with a blurred background exactly?
If a shape had a blurred background, everything behind it would still look unblurred if the shape is filled as a color.
If you can still fill up the shape with a pattern made of a blurred image.

**Comment by abhaybisht314ecorp** [comment]
<img width="233" alt="Screenshot 2024-12-16 at 10 19 48 AM" src="https://github.com/user-attachments/assets/3d6a375f-99b2-4ad9-8721-45ed888fd671" />
@asturur i want to achieve this with fabric.js.

**Comment by asturur** [comment]
Yes blurring the shape background will do nothing. you have to make a pattern in the shape with the image behind it, and blur the image used for the pattern, then update the pattern offset to reflect the shape moving.
Is a complex effect and can be  also not feasible depending on how you want to use it
